### ISSUE TITLE
Should gluon layer support **name** field?

### ISSUE THREAD

**Issue by yangfly** [apache/incubator-mxnet#14515]
At present, the **name** of **Gluon** layer is automatically extended to `xxx/operator0_fwd`, based on `name_scope`, which embodies modularization and avoids naming conflicts. However, the **name** field is not currently supported, resulting in a lack of flexibility. 

for example
```
gluon.nn.Conv2D(10, 3)  # xxx/conv0_fwd
```
Should **Gluon** layer also support custom **name** like `Symbol`. Internally, only layer without user defined `name` are automatically named. 
```
gluon.nn.Conv2D(10, 3, name='conv1')
```
This feature is useful in the following scenarios
- `Gluon` models can be aligned with `Symbol` models or other framework, and users can specify more consistent naming method.
- `Gluon` models can be compared more intuitively with the net structure in the papers with `viz.print_summary` or `viz.plot_network`.
- `Symbol` can be load more conveniently from `Gluon` with `load_parameters`.

Welcome to this discussion!

**Comment by wkcn** [comment]
+1. I think it is necessary to support custom name for gluon block.
For example, to implement a ResNet model, I would like to name the layers as the paper names. In addition, I have a pre-trained model for MXNet Symbol, and I need the consistent names of parameters to support Gluon Model.

**Comment by eureka7mt** [comment]
You can use
`gluon.nn.Conv2D(10, 3, prefix='conv1_')`

**Comment by yangfly** [comment]
@eureka7mt Thank you for your advice, but there are two drawbacks with `prefix=name_`. 
- Naming conflicts: conv with prefix='conv1_' and the 2th auto-named conv. 
- Unnecessary suffix: `_fwd`

![image](https://user-images.githubusercontent.com/15797180/55123521-f06c6000-513d-11e9-9aa7-f55d3751ca22.png)

- Perhaps we should set a pythonic prefix '_' for auto-named layers, like `_conv1`, `_pool1`, `_batchnorm1`, which is different from user defined names `conv1`, `pool1` and `batchnorm1`。And why `_fwd` is necessary in auto-naming?

**Comment by djaym7** [comment]
following, request +1
